### PR TITLE
Automatic CLI login

### PR DESCRIPTION
**Pull request by RamiroND**
New blog post.

### CODE REVIEW

File: apps/www/_blog/2023-12-01-automatic-cli-login.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+title: 'Automatic CLI login'
+description: 'Explore the technical implementation and security measures behind CLI's new automatic login feature.'
```
Comment (fsansalvadore) [inline apps/www/_blog/2023-12-01-automatic-cli-login.mdx:3]: The sneaky problem here is that the `'` in CLI's conflicts with the opening and closing `'`s.
The solution is to open and close with `"`. @RamiroND

**Review by fsansalvadore — APPROVED** [review]

File: apps/www/lib/authors.json [hunk 2]
```diff
@@ -358,5 +358,13 @@
     "position": "Engineering",
     "author_url": "https://twitter.com/beertocode",
     "author_image_url": "https://github.com/loong.png"
+  },
+  {
+    "author_id": "kamil_ogorek",
+    "author": "Kamil Ogorek",
```
Comment (kamilogorek) [inline apps/www/lib/authors.json:364]:
```suggestion
    "author": "Kamil Ogórek",
```
🇵🇱

File: apps/www/_blog/2023-12-01-automatic-cli-login.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+After gathering all this data, a unique URL is generated containing the session ID, token name, and your public key. The CLI then attempts to open this URL in your current environment's default browser. Once it opens, it verifies your authentication. If you're not authenticated, you're redirected to the login page, prompted to log into the dashboard, and upon successful authentication, redirected back to the originally requested page.
+
+With your identity confirmed, the CLI sends the provided session data to our management API, requesting its magic.
+
+The Management API generates a new access token for your account, creates its pair of ECDH keys, and encrypts the token using your public key and the server's private key. This encrypted token, alongside the server's public key, your session ID, and expiration date, is then stored in our database.
+
+The Management API generates a new access token for your account. To send it back to your CLI it needs to be stored in a database.
+
+## Token encryption with Elliptic-Curve Diffie-Hellman
```
Comment (hf) [inline apps/www/_blog/2023-12-01-automatic-cli-login.mdx:45]: Maybe:

```suggestion
## Securely exchanging the token using ECDH
```

Technically the encryption is not ECDH -- ECDH is just the method to agree on a shared encryption key. I have a feeling HN will point this out. 😂

**Review by hf — COMMENTED** [review]
OK!

**Pull request merged**